routes/post.go

```diff
@@ -1599,17 +1600,601 @@ func (fes *APIServer) GetDiamondedPosts(ww http.ResponseWriter, req *http.Reques
+			DiamondLevel:         pkidToDiamondLevel[*diamondSenderPKID.PKID],
+		}
+		diamondSenderResponses = append(diamondSenderResponses, diamondSenderResponse)
+	}
+
+	// Return the posts found.
+	res := &GetDiamondsForPostResponse{
+		DiamondSenders: diamondSenderResponses,
+	}
+	if err := json.NewEncoder(ww).Encode(res); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetDiamondsForPost: Problem encoding response as JSON: %v", err))
+		return
+	}
+}
+
+type GetRecloutsForPostRequest struct {
+	// PostHashHex to fetch.
+	PostHashHex                string `safeForLogging:"true"`
+	Offset                     uint32 `safeForLogging:"true"`
+	Limit                      uint32 `safeForLogging:"true"`
+	ReaderPublicKeyBase58Check string `safeForLogging:"true"`
+}
+
+type GetRecloutsForPostResponse struct {
+	Reclouters []*ProfileEntryResponse
+}
+
+func (fes *APIServer) GetRecloutsForPost(ww http.ResponseWriter, req *http.Request) {
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := GetRecloutsForPostRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetRecloutsForPost: Problem parsing request body: %v", err))
+		return
+	}
+
+	// Decode the postHash.
+	var postHash *lib.BlockHash
+	if requestData.PostHashHex == "" {
+		_AddBadRequestError(ww, fmt.Sprintf("GetRecloutsForPost: Must provide a PostHashHex to fetch."))
+		return
+	} else {
+		postHashBytes, err := hex.DecodeString(requestData.PostHashHex)
+		if err != nil || len(postHashBytes) != lib.HashSizeBytes {
+			_AddBadRequestError(ww, fmt.Sprintf("GetRecloutsForPost: Error parsing post hash %v: %v",
+				requestData.PostHashHex, err))
+			return
+		}
+		postHash = &lib.BlockHash{}
+		copy(postHash[:], postHashBytes)
+	}
+
+	// Decode the reader public key into bytes. Default to nil if no pub key is passed in.
+	var readerPublicKeyBytes []byte
+	if requestData.ReaderPublicKeyBase58Check != "" {
+		var err error
+		readerPublicKeyBytes, _, err = lib.Base58CheckDecode(requestData.ReaderPublicKeyBase58Check)
+		if requestData.ReaderPublicKeyBase58Check != "" && err != nil {
+			_AddBadRequestError(ww, fmt.Sprintf("GetRecloutsForPost: Problem decoding user public key: %v : %s", err, requestData.ReaderPublicKeyBase58Check))
+			return
+		}
+	}
+
+	// Get a view with all the mempool transactions.
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetRecloutsForPost: Error constucting utxoView: %v", err))
+		return
+	}
+
+	// Fetch the reclouters for the post requested.
+	reclouterPubKeys, err := utxoView.GetRecloutsForPostHash(postHash)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetRecloutsForPost: Error getting reclouters %v", err))
+		return
+	}
+
+	// Filter out any restricted profiles.
+	addReaderPublicKey := false
```

Comment:
I think we can get `addReaderPublicKey` in a similar fashion to my comment on the `GetLikesForPost` about using `GetPostEntryReaderState` instead of comparing every public key to the reader